corgi/tasks/pnc.py

```diff
@@ -30,4 +36,82 @@ def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
     response.raise_for_status()
     logger.info(f"SBOM fetched for purl {purl}")
 
-    parse_pnc_sbom(response.json())
+    # Make sure the SBOM is valid
+    sbom = SbomerSbom(response.json()["sbom"])
+
+    # Create a build for the root component
+    root_build, created = SoftwareBuild.objects.get_or_create(
+        build_id=sbom.components["root"]["meta_attr"]["pnc_build_id"],
+        build_type=SoftwareBuild.Type.PNC,
+    )
+
+    # Create ProductComponentRelation if it doesn't already exist
+    if created:
+        pcr, _ = ProductComponentRelation.objects.get_or_create(
+            build_id=root_build.build_id,
+            build_type=SoftwareBuild.Type.PNC,
+            software_build__isnull=root_build,
+            defaults={
+                "type": ProductComponentRelation.Type.SBOMER,
+            },
+        )
+
+    # Create components
+    components = {}
+    for bomref, component in sbom.components.items():
+        defaults = {"namespace": component["namespace"], "meta_attr": component["meta_attr"]}
+
+        if "related_url" in component:
+            defaults["related_url"] = component["related_url"]
+        if bomref == "root":
+            defaults["software_build"] = root_build
+
+        components[bomref], _ = Component.objects.update_or_create(
+            type=Component.Type.MAVEN,
+            name=component["name"],
+            version=component["version"],
+            release="",
+            arch="noarch",
+            defaults=defaults,
+        )
+
+    # Link dependencies
+    nodes = {}
+    # Create the root node from the SBOM's component
+    root_component = components.pop("root")
+    nodes["root"], _ = ComponentNode.objects.get_or_create(
+        type=ComponentNode.ComponentNodeType.SOURCE,
+        parent=None,
+        defaults={"obj": root_component},
+    )
+
+    # Create the relationships recorded in the manifest's dependencies
+    for parent, deps in sbom.dependencies.items():
+        # Only create a new node for parent if it wasn't already
+        # created as a dependency of another component
+        if parent not in nodes:
+            nodes[parent], _ = ComponentNode.objects.get_or_create(
+                type=ComponentNode.ComponentNodeType.PROVIDES,
```

Comment:
There's currently no distinction between PROVIDES and PROVIDES_DEV in the CycloneDX manifest. It's tough to say whether a distinction could be made, given Quarkus' nature as a development framework itself, but I'll see if we can more accurately identify those components.

Regarding upstreams, that's _also_ a tough question, since the root components are essentially just metadata files themselves.